src/NuGet.Clients/NuGet.CommandLine/Common/SelfUpdater.cs

```diff
@@ -96,7 +105,7 @@ internal async Task UpdateSelfFromVersionAsync(string exePath, bool prerelease,
                     {
                         DirectoryUtility.CreateSharedDirectory(tempDir);
 
-                        DownloadResourceResult downloadResourceResult = await PackageDownloader.GetDownloadResourceResultAsync(
+                        using DownloadResourceResult downloadResourceResult = await PackageDownloader.GetDownloadResourceResultAsync(
```

Comment:
On CI, `WhenPackageDownloadedFromNuGetOrgAuthenticodeIsVerified_SucceedsAsync` test with the following error.

`NuGet.Protocol.Core.Types.FatalProtocolException : Error downloading 'NuGet.CommandLine.6.0.0' from 'https://api.nuget.org/v3-flatcontainer/nuget.commandline/6.0.0/nuget.commandline.6.0.0.nupkg'.\r\n---- System.IO.IOException : The process cannot access the file 'nuget.commandline.6.0.0.nupkg' because it is being used by another process.`
Upon checking the CI logs, I learnt that `WhenPackageDownloadedFromNuGetOrgAuthenticodeIsVerified_FailsAsync` test executed first but the package folder in the temporary directory failed to delete because the file is in use. Upon looking at the source code, I learnt that `DownloadResourceResult` implements `IDisposable` interface. I assume this class is holding a file handle to the package in the temporary folder.  Hence added a `using` statement so that `dispose` method closes all file handle. The folder delete operation following this code should succeed.